cos-fleetshard-operator/src/main/java/org/bf2/cos/fleetshard/operator/connector/ConnectorConfiguration.java

```diff
@@ -42,7 +42,7 @@ public ConnectorConfiguration(ObjectNode connectorSpec, Class<S> type) {
             LOGGER.error("Missing `data_shape` in connector spec!");
             this.dataShapeSpec = null;
         } else {
-            this.dataShapeSpec = dataShape.deepCopy();
+            this.dataShapeSpec = Serialization.unmarshal(dataShape.deepCopy().toString(), dataShapeType);
```

Comment:
wonder if a deep copy is really needed

Reply:
I think it isn't needed anymore when we deserialize, true! I will upate this part!